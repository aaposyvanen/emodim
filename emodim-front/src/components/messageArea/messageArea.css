.thread-start .message-box {
    margin: 0;
    border: 0;
}

.comments {
    margin-left: 10px;
    padding-left: 10px;
    border-left: 3px solid #B7B7B7;
}

.replies {
    margin-left: 10px;
    padding-left: 10px;
    border-left: 3px solid #B7B7B7;
}

.response-section {
    display: flex;
    flex-flow: column;
}

.author {
    font-weight: bold;
}