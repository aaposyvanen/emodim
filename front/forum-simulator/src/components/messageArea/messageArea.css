
.thread-start {
    margin: 5px;
    padding: 5px;
    background-color: #202020;
    border-bottom: 2px black solid;
}

.thread-start .message-box {
    margin: 0;
    border-bottom: 0;
}

.comments {
    margin: 5px 0px 5px 20px;
    padding: 5px;
}
