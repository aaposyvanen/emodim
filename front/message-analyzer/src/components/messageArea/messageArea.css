
.thread-start {
    margin: 5px;
    padding: 5px;
    background-color: #202020;
    border-radius: 10px 10px 0 0 ;
}

.thread-start .message-box {
    margin: 0;
    border: 0;
}

.response-section {
    display: flex;
    flex-flow: column;
    margin: 10px 5px 5px;
    border-radius: 10px 10px 0 0;
    background-color: #202020;
}

.comments {
    margin: 5px 0px 5px 20px;
    padding: 5px;
}
