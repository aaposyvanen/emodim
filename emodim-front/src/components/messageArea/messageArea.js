import React, { useEffect, useRef } from "react";
import { useSelector, useDispatch } from "react-redux";
import _ from "lodash";
import "./messageArea.css";
import AnnotatedMessage from "../annotatedMessage/annotatedMessage";
import Comment from "../comment/comment";
import ResponseSection from "../responseSection/responseSection";
import { updateCurrentThread } from "../../actions/threadActions";
import { separateResponsesFromComments, moveResponsesToTheirParents} from "../app"
import SaveThread from "../saveThread/saveThread";
import { chatEndpoint } from "../../constants";
import socketIOClient from "socket.io-client";
import { formWordArrayFromAnalyzedData } from "../responseAnalysisDialog/responseAnalysisDialog";
import { addMessageToCurrentThread } from "../../actions/threadActions";

export const MessageArea = () => {
    const dispatch = useDispatch();
    const socketRef = useRef(null);

    const startMessage = useSelector(state => state.threadReducer.thread.startMessage); 
    const currentAnnotations = useSelector( state => state.annotationsReducer.annotations);
    const currentThread = useSelector(state => state.threadReducer.thread);
    const currentComments = useSelector(state => state.threadReducer.thread.comments);
    
    useEffect(() => {
        const socket = socketIOClient(chatEndpoint);
        socketRef.current = socket;

        socket.on("message", message => {
            message.words = formWordArrayFromAnalyzedData(message.words);
            dispatch(addMessageToCurrentThread(message));
        });

        return () => socket.disconnect();
    }, [dispatch]);

    useEffect(() => {
        if (currentThread && currentComments) {
            const [comments, responses] = separateResponsesFromComments(currentThread.comments);
            const commentsWithResponses = moveResponsesToTheirParents(comments, responses);
            currentThread.comments = commentsWithResponses;
            dispatch(updateCurrentThread(currentThread));
        }
    }, [currentThread, dispatch]);

    return (
        <div className="message-area">
            <div className="thread-start">
                {
                    startMessage.commentMetadata
                        ?
                        <AnnotatedMessage
                            data={startMessage}
                            wordLevelAnnotations={currentAnnotations.message.wordHighlights}
                            messageLevelAnnotations={currentAnnotations.message.messageAnalysis}
                            emoji={currentAnnotations.message.emoji}
                            sidebar={currentAnnotations.message.sidebar}
                            response={true}
                        />
                        :
                        null
                }
            </div>

            <div className="comments">
                {
                    currentThread.comments && currentThread.comments.map(comment => {
                        if (comment && comment.commentMetadata) {
                            return (
                                <div key={`div-${comment.commentMetadata.comment_id}`}>
                                    <Comment
                                        data={comment}
                                        key={comment.commentMetadata.comment_id}
                                        wordLevelAnnotations={currentAnnotations.message.wordHighlights}
                                        messageLevelAnnotations={currentAnnotations.message.messageAnalysis}
                                        emoji={currentAnnotations.message.emoji}
                                        sidebar={currentAnnotations.message.sidebar}
                                    />
                                    {comment.children && !_.isEmpty(comment.children)
                                        ? <div className="replies">
                                            {
                                                comment.children.map(child => {
                                                    return <AnnotatedMessage
                                                        data={child}
                                                        key={child.commentMetadata.comment_id}
                                                        response
                                                        wordLevelAnnotations={currentAnnotations.message.wordHighlights}
                                                        messageLevelAnnotations={currentAnnotations.message.messageAnalysis}
                                                        emoji={currentAnnotations.message.emoji}
                                                        sidebar={currentAnnotations.message.sidebar}
                                                    />
                                                })
                                            }
                                         </div>
                                        : null
                                    }
                                </div>
                            )
                        } else {
                            return null;
                        }
                    })
                }
            </div>
            <div className="response-section">
                <ResponseSection 
                    wordLevelAnnotations={currentAnnotations.feedback.wordHighlights}
                    messageLevelAnnotations={currentAnnotations.feedback.messageAnalysis}
                    toggleResponsefield={null}
                    commentId={0}
                />
            </div>
            <SaveThread/>
        </div>
    );
}

export default MessageArea;
