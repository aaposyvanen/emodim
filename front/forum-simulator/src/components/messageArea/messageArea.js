import React, { Component } from "react";
import { connect } from "react-redux";

import "./messageArea.css";
import Message from "../message/message";

export class MessageArea extends Component {

    render() {
        return (
            <div className="message-area">
                <div className="thread-start">
                    <Message
                        data={this.props.startMessage}
                        styleWords={false}
                    />
                </div>
                <div className="comments">
                    {
                        this.props.comments && this.props.comments.map(comment => {
                            return < Message
                                data={comment}
                                key={comment.commentMetadata.id}
                                styleWords={false}
                            />
                        })
                    }
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        startMessage: state.threadReducer.thread.startMessage,
        comments: state.threadReducer.thread.comments
    };
}


export default connect(mapStateToProps)(MessageArea);
