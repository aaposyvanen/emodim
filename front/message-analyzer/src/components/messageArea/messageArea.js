import React, { Component } from "react";
import { connect } from "react-redux";

import "./messageArea.css";
import Message from "../message/message";
import ResponseField from "../responseField/responseField";
import ResponseAnalysisDialog from "../responseAnalysisDialog/responseAnalysisDialog";

export class MessageArea extends Component {

    render() {
        return (
            <div className="message-area">
                <div className="thread-start">
                    <Message
                        data={this.props.startMessage}
                    />
                </div>
                <div className="response-section">
                    <ResponseField />
                    <ResponseAnalysisDialog />
                </div>
                <div className="comments">
                    {
                        this.props.comments && this.props.comments.map(comment => {
                            return < Message
                                data={comment}
                                key={comment.commentMetadata.id}
                            />
                        })
                    }
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        startMessage: state.threadReducer.thread.startMessage,
        comments: state.threadReducer.thread.comments
    };
}


export default connect(mapStateToProps)(MessageArea);
