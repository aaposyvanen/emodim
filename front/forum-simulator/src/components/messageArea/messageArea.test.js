import React from 'react';
import { render } from "@testing-library/react"
import { MessageArea } from "./messageArea";

test("MessageArea renders correctly", () => {
    const { container } = render(<MessageArea />);
    expect(container).toMatchSnapshot();
});
