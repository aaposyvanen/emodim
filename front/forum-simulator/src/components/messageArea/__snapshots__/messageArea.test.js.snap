// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageArea renders correctly with nested comments 1`] = `
<div>
  <div
    class="message-area"
  >
    <div
      class="thread-start"
    >
      <div
        class="message-box"
      >
        <div
          class="metadata"
        >
          <div
            class="author"
          >
            Author1
          </div>
          <div
            class="date"
          >
            2000-01-01 00:01:00
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            <span>
               
            </span>
            <span
              class="arousal0 valence0"
            >
              test1
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="comments"
    >
      <div
        class="message-box"
      >
        <div
          class="metadata"
        >
          <div
            class="author"
          >
            Author2
          </div>
          <div
            class="date"
          >
            2000-01-01 00:02:00
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            <span>
               
            </span>
            <span
              class="arousal0 valence0"
            >
              test2
            </span>
          </span>
        </div>
        <div
          class="message-box response"
        >
          <div
            class="metadata"
          >
            <div
              class="author"
            >
              Author4
            </div>
            <div
              class="date"
            >
              2000-01-01 00:04:00
            </div>
          </div>
          <div
            class="message"
          >
            <span>
              <span>
                 
              </span>
              <span
                class="arousal0 valence0"
              >
                test4
              </span>
            </span>
          </div>
        </div>
        <div
          class="message-box response"
        >
          <div
            class="metadata"
          >
            <div
              class="author"
            >
              Author5
            </div>
            <div
              class="date"
            >
              2000-01-01 00:05:00
            </div>
          </div>
          <div
            class="message"
          >
            <span>
              <span>
                 
              </span>
              <span
                class="arousal0 valence0"
              >
                test5
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="message-box"
      >
        <div
          class="metadata"
        >
          <div
            class="author"
          >
            Author3
          </div>
          <div
            class="date"
          >
            2000-01-01 00:03:00
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            <span>
               
            </span>
            <span
              class="arousal0 valence0"
            >
              test3
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MessageArea renders correctly with props 1`] = `
<div>
  <div
    class="message-area"
  >
    <div
      class="thread-start"
    >
      <div
        class="message-box"
      >
        <div
          class="metadata"
        >
          <div
            class="author"
          >
            Author
          </div>
          <div
            class="date"
          >
            2000-01-01 00:00:00
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            <span>
               
            </span>
            <span
              class="arousal0 valence0"
            >
              test
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="comments"
    >
      <div
        class="message-box"
      >
        <div
          class="metadata"
        >
          <div
            class="author"
          >
            Author
          </div>
          <div
            class="date"
          >
            2000-01-01 00:00:00
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            <span>
               
            </span>
            <span
              class="arousal0 valence0"
            >
              test1
            </span>
          </span>
        </div>
      </div>
      <div
        class="message-box"
      >
        <div
          class="metadata"
        >
          <div
            class="author"
          >
            Author
          </div>
          <div
            class="date"
          >
            2000-01-01 00:00:00
          </div>
        </div>
        <div
          class="message"
        >
          <span>
            <span>
               
            </span>
            <span
              class="arousal0 valence0"
            >
              test2
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MessageArea renders correctly without props 1`] = `
<div>
  <div
    class="message-area"
  >
    <div
      class="thread-start"
    />
    <div
      class="comments"
    />
  </div>
</div>
`;
